app/src/pages/playground/playgroundUtils.ts

```diff
@@ -815,11 +819,16 @@ export const getChatCompletionVariables = ({
   let invocationParameters: InvocationParameterInput[] = [
     ...instance.model.invocationParameters,
   ];
+  const convertedToolChoice = safelyConvertToolChoiceToProvider({
+    toolChoice: instance.toolChoice,
+    targetProvider: instance.model.provider,
+  });
   if (instance.tools.length > 0) {
-    invocationParameters.push({
-      invocationName: TOOL_CHOICE_PARAM_NAME,
-      valueJson: instance.toolChoice,
-    });
+    invocationParameters = invocationParameters.map((param) =>
+      param.canonicalName === TOOL_CHOICE_PARAM_CANONICAL_NAME
+        ? { ...param, valueJson: convertedToolChoice }
+        : param
+    );
```

Comment:
does this get filtered out in yoru other pr? i know for openai we have to have this, is this guaranteed to add it every time?